src/vaults/SlippageVault.sol

```diff
@@ -0,0 +1,231 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.16;
+
+import {MathUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/math/MathUpgradeable.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+
+import {Vault} from "src/vaults/Vault.sol";
+import {VaultV2} from "src/vaults/VaultV2.sol";
+import {NftGate} from "src/vaults/NftGate.sol";
+import {HarvestStorage} from "src/vaults/HarvestStorage.sol";
+import {BaseStrategy as Strategy} from "src/strategies/BaseStrategy.sol";
+import {uncheckedInc} from "src/libs/Unchecked.sol";
+import {VaultErrors} from "src/libs/VaultErrors.sol";
+
+contract SlippageVault is VaultV2 {
+    using SafeTransferLib for ERC20;
+    using MathUpgradeable for uint256;
+
+    /**
+     * @dev See {IERC4262-deposit}.
+     */
+    function deposit(uint256 assets, address receiver) public virtual override whenNotPaused returns (uint256) {
+        _harvestAll();
```

Comment:
Instead of calling`_harvestAll()`, we should be getting the total strategy holdings before/after we deposit into the strategies. The difference between these two numbers is what should be passed to `previewDeposit`